### PR TITLE
Add vsock supporting on s390x

### PR DESCRIPTION
**Pull request by fbq815**
vsock:Add suporting for vsock on s390x,modify the device_pattern works
on s390x

ID:1937706

Signed-off-by: Boqiao Fu <bfu@redhat.com>

### CODE REVIEW

**Comment by fbq815** [comment]
Test result:
JOB ID     : f45be59983f88cc4c63b78477290d84561eb6ed2
JOB LOG    : /home/bfu/kar/workspace/job-results/job-2021-03-11T06.56-f45be59/job.log
 (1/1) Host_RHEL.m8.u4.product_av.nographic.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.4.0.s390x.io-github-autotest-qemu.vsock_test.hotplug.s390-virtio: PASS (127.97 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/bfu/kar/workspace/job-results/job-2021-03-11T06.56-f45be59/results.html
JOB TIME   : 128.70 s

**Comment by fbq815** [comment]
@PaulYuuu @zhenyzha @ngu-niny please help me review this patch

**Comment by zhenyzha** [comment]
Because other platforms have product bugs, please do not add them in the public area.
And it is recommended to create a separate test case outside of basic_test.

File: qemu/tests/cfg/vsock.cfg [hunk 1]
```diff
@@ -2,6 +2,7 @@
     no Windows
     type = vsock_test
     vsocks = vhost_vsock0
+    virtio_dev_iommu_platform = on
```
Comment (PaulYuuu) [inline qemu/tests/cfg/vsock.cfg:5]: Remove this line since it‘s personal usage.

Comment (fbq815) [reply qemu/tests/cfg/vsock.cfg]: solved

File: qemu/tests/vsock_hotplug.py [hunk 2]
```diff
@@ -48,27 +49,29 @@ def run(test, params, env):
                           'dmesg output.', logging.info)
     addr_pattern = params['addr_pattern']
     device_pattern = params['device_pattern']
-    lspci_cmd = 'lspci'
+    vsock_check_cmd = params.get('lspci_cmd', 'lspci')
```
[the remark this answers is not in the document]
Comment (fbq815) [reply qemu/tests/vsock_hotplug.py:52]: fixed with check_vsock_cmd

File: qemu/tests/vsock_hotplug.py [hunk 3]
```diff
@@ -40,6 +40,7 @@ def run(test, params, env):
     if '-mmio:' in params.get('machine_type'):
         dev_vsock = qdevices.QDevice('vhost-vsock-device', vsock_params)
     elif params.get('machine_type').startswith("s390"):
+        vsock_params['devno'] = 'fe.1.2345'
```
[the remark this answers is not in the document]
Comment (fbq815) [reply qemu/tests/vsock_hotplug.py:43]: move value to cfg

File: qemu/tests/cfg/vsock.cfg [hunk 4]
```diff
@@ -2,6 +2,8 @@
     no Windows
     type = vsock_test
     vsocks = vhost_vsock0
+    s390x:
+        virtio_dev_iommu_platform = on
```
Comment (PaulYuuu) [inline qemu/tests/cfg/vsock.cfg:6]: If s390x can launch a guest vsock without `virtio_dev_iommu_platform = on`, make it consistent with the behavior of other platforms.

File: qemu/tests/vsock_hotplug.py [hunk 5]
```diff
@@ -48,27 +49,29 @@ def run(test, params, env):
                           'dmesg output.', logging.info)
     addr_pattern = params['addr_pattern']
     device_pattern = params['device_pattern']
-    lspci_cmd = 'lspci'
+    check_vsock_cmd = params.get('lspci_cmd', 'lspci')
```
Comment (PaulYuuu) [inline qemu/tests/vsock_hotplug.py:52]: `check_vsock_cmd` & `lspci_cmd`, keep the namespace consistent.

Comment (fbq815) [reply qemu/tests/vsock_hotplug.py]: solved

File: qemu/tests/cfg/vsock.cfg (hunk 4 above)
Comment (fbq815) [reply qemu/tests/cfg/vsock.cfg]: removed

**Comment by ngu-niny** [comment]
LGTM

File: qemu/tests/vsock_hotplug.py [hunk 6]
```diff
@@ -48,27 +49,29 @@ def run(test, params, env):
...
-    else:
-        address = re.findall(addr_pattern, device_str[0])[0]
-        chk_dmesg_cmd = 'dmesg'
-        output = re.findall(address, session.cmd_output(chk_dmesg_cmd))
-        if not output:
-            test.fail('dmesg failed, no info related to %s' % address)
+
+    if params.get('dmesg_check') == 'yes':
+        if not device_str:
+            test.fail('lspci failed, no device "%s"' % device_pattern)
```
Comment (PaulYuuu) [inline qemu/tests/vsock_hotplug.py:60]: Please also update the fail message.

Comment (fbq815) [reply qemu/tests/vsock_hotplug.py]: modified

**Comment by PaulYuuu** [comment]
@fbq815 Since you remove iommu related configuration, please update the commit accordingly.

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
**AGAIN**, limit 72 char per line on git commit. https://avocado-framework.readthedocs.io/en/86.0/guides/contributor/chapters/styleguides.html
```
header          <- Limited to 72 characters. No period.
                <- Blank line
message         <- Any number of lines, limited to 72 characters per line.
                <- Blank line
Reference:      <- External references, one per line (issue, trello, ...)
Signed-off-by:  <- Signature and acknowledgment of licensing terms when
                   contributing to the project (created by git commit -s)
```

**Comment by PaulYuuu** [comment]
> **AGAIN**, limit 72 char per line on git commit. https://avocado-framework.readthedocs.io/en/86.0/guides/contributor/chapters/styleguides.html
> 
> ```
> header          <- Limited to 72 characters. No period.
>                 <- Blank line
> message         <- Any number of lines, limited to 72 characters per line.
>                 <- Blank line
> Reference:      <- External references, one per line (issue, trello, ...)
> Signed-off-by:  <- Signature and acknowledgment of licensing terms when
>                    contributing to the project (created by git commit -s)
> ```

Made a mistake myself.

**Pull request merged**